python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -661,6 +870,166 @@ def asset_check_spec(self) -> AssetCheckSpec:
+        """Deprecated. Use partition_key_range property or partition_key_range_for_asset instead.
+
+        Return the PartitionKeyRange for the corresponding output. Errors if not present.
+        """
+        asset_key = self.asset_key_for_output(output_name)
+        return self.partition_key_range_for_asset(asset_key)
+
+    @deprecated(
+        breaking_version="2.0", additional_warn_text="Use `partition_key_range_for_asset` instead."
+    )
+    @public
+    def asset_partition_key_range_for_input(self, input_name: str) -> PartitionKeyRange:
+        """Deprecated. Use partition_key_range_for_asset instead.
+
+        Return the PartitionKeyRange for the corresponding input. Errors if there is more or less than one.
+        """
+        asset_key = self.asset_key_for_input(input_name)
+        return self.partition_key_range_for_asset(asset_key, is_dependency=True)
+
+    @deprecated(
+        breaking_version="2.0",
+        additional_warn_text="Use `partitions_def` or `partitions_def_for_asset` instead.",
+    )
+    @public
+    def asset_partitions_def_for_output(self, output_name: str = "result") -> PartitionsDefinition:
+        """Deprecated. Use partitions_def property or partitions_def_for_asset instead.
+
+        The PartitionsDefinition on the upstream asset corresponding to this input.
+        """
+        asset_key = self.asset_key_for_output(output_name)
+        return self.partitions_def_for_asset(asset_key)
+
+    @deprecated(
+        breaking_version="2.0", additional_warn_text="Use `partitions_def_for_asset` instead."
+    )
+    @public
+    def asset_partitions_def_for_input(self, input_name: str) -> PartitionsDefinition:
+        """Deprecated. Use partitions_def_for_asset instead.
+
+        The PartitionsDefinition on the upstream asset corresponding to this input.
+        """
+        asset_key = self.asset_key_for_input(input_name)
+        return self.partitions_def_for_asset(asset_key)
+
+    @deprecated(
+        breaking_version="2.0", additional_warn_text="Use `partition_keys_for_asset` instead."
+    )
+    @public
+    def asset_partition_keys_for_output(self, output_name: str = "result") -> Sequence[str]:
+        """Deprecated. Use partition_keys_for_asset instead.
+
+        Returns a list of the partition keys for the given output.
+        """
+        asset_key = self.asset_key_for_output(output_name)
+        return self.partition_keys_for_asset(asset_key)
+
+    @deprecated(
+        breaking_version="2.0", additional_warn_text="Use `partition_keys_for_asset` instead."
+    )
+    @public
+    def asset_partition_keys_for_input(self, input_name: str) -> Sequence[str]:
+        """Deprecated. Use partition_keys_for_asset instead.
+
+        Returns a list of the partition keys of the upstream asset corresponding to the
+        given input.
+        """
+        asset_key = self.asset_key_for_input(input_name)
+        return self.partition_keys_for_asset(asset_key, is_dependency=True)
```

Comment:
unclear that we will want to deprecate these as long as `AssetOut` / `AssetIn` world is still supported. Since you define your objects in terms of output names there and the keys are derived for you it seem less than ideal to force users to translate

Reply:
Agree with alex. We're aren't completely divorcing ourselves from I/O manager, but we are making them more more "opt-in" and making non-io-manager land feel more first-class. For io-manager-forward users, this is a big regression.

Reply:
ok. My main concern with keeping all of these different versions around is how confusing all of the names will become. Seeing three to four variations of a method (i.e. `asset_partitions_keys_for_input()`, `asset_partition_keys_for_output()`, `partition_keys_for_asset()`, and `partition_keys` property) would be so daunting and would require the user to learn about each of those methods before they can pick the correct one. 

We could probably resolve some of this with good API docs:
* In the doc block for each method, indicate if it is intended for use with I/O managers. Maybe not in those words exactly, but give the user the ability to easily know if this method is relevant to them.
* Manually create sections in the API docs page that groups the "for_asset" methods together, the "for_output" methods together, and the "for_input" methods together. I think this kind of manual ordering of methods within a class is possible in sphinx, but I'm not 100% sure
* If we can do ^ type of manual ordering, we could present the methods in a way that goes from least complex to more complex. Right now it automatically sorts of the methods of the `context` alphabetically, so if you scroll through you are presented with the methods in a non-sensical order